jupyter_server/serverapp.py

```diff
@@ -1886,9 +1887,9 @@ def init_configurables(self):
             connection_dir=self.runtime_dir,
             kernel_spec_manager=self.kernel_spec_manager,
         )
+        self.file_id_manager = FileIdManager(parent=self, log=self.log)
```

Comment:
Good catch! I forgot to define `FileIdManager` as a trait on `ServerApp`. Before I fix this, is there any reason to prefer

```
self.file_id_manager_class = Type(klass=FileIdManager)
...
self.file_id_manager = self.file_id_manager_class(*args)
```

over just using

```
self.file_id_manager = Instance(klass=FileIdManager, args=(*args))
```

Reply:
Neither is right.

Both imply that one has to derive from `FileIdManager` - which we shouldn't impose on anyone.  So `klass` should reference an ABC (e.g., `FileIdManagerABC`) that only defines the public APIs as abstract.  It could also contain validation logic for the db-path trait, etc.

So the user would invoke the server using `--ServerApp.file_id_manager_class = my.package.MyFileIdManager` and the trait would ensure that `my.package.MyFileIdManager` is an instance of `FileIdManagerABC` and whatever interacts with the file ID manager class would be relegated to the methods defined in the ABC.

Folks that want to slightly adjust the OOTB implementation of `FileIdManager` (or override all of its methods entirely) are free to do so via subclassing as they are still an instance of the ABC.

For example, with the last set of changes, an entirely different implementation is required if any non-filesystem-based `ContentsManager` is configured (unfortunately).